handshake/crypto_setup_client.go

```diff
@@ -83,30 +85,51 @@ func NewCryptoSetupClient(
 		nullAEAD:             crypto.NewNullAEAD(protocol.PerspectiveClient, version),
 		aeadChanged:          aeadChanged,
 		negotiatedVersions:   negotiatedVersions,
+		divNonceChan:         make(chan struct{}),
 	}, nil
 }
 
 func (h *cryptoSetupClient) HandleCryptoStream() error {
+	messageChan := make(chan HandshakeMessage)
+
+	go func() {
+		for {
+			message, err := ParseHandshakeMessage(h.cryptoStream)
+			if err != nil {
+				close(messageChan)
+				return
```

Comment:
This silently discards the error here. Either use this error as the return value of the function (which is notoriously tricky with go channels 🙄), or log the error.